Extend/Model/PaymentGateway.php

```diff
@@ -349,9 +337,14 @@ private static function _addBasketInfo(&$oTransaction, $oBasket)
      */
     public static function getModuleToken($oSession)
     {
```

Comment:
getModueToken is also saving the token in the session, that is not single responsibility. Please rename the method or refactor it to two methods depending on what is better.

Reply:
I will rename it, it is mandatory to set it and get the same token back.